File: packages/evals/tasks/bench/act/os_dropdown.ts
import { defineBenchTask } from "../../../framework/defineTask.js";

export default defineBenchTask(
  { name: "os_dropdown" },
  async ({ debugUrl, sessionUrl, stagehand, page, logger }) => {
    /**
     * This eval is meant to test whether we can correctly select an element
     * from an OS level dropdown
     */

    try {
      await page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/nested-dropdown/");

      await stagehand.act("choose 'Smog Check Technician' from the 'License Type' dropdown");
      const selectedOption = await page.evaluate(() => {
        const option = document.querySelector(
          "#licenseType option:checked",
        ) as HTMLOptionElement | null;
        return option?.textContent ?? null;
      });

      if (selectedOption === "Smog Check Technician") {
        return {
          _success: true,
          debugUrl,
          sessionUrl,
          logs: logger.getLogs(),
        };
      }
      return {
        _success: false,
        message: "incorrect option selected from the dropdown",
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    } catch (error) {
      return {
        _success: false,
        error: String(error),
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    } finally {
      await stagehand.close();
    }
  },
);

File: packages/sdk-ruby/evals/tasks/act/os_dropdown.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/os_dropdown.ts

Evals.define_task("os_dropdown") do |t|
  # This eval is meant to test whether we can correctly select an element
  # from an OS level dropdown

  t.page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/nested-dropdown/")

  t.stagehand.act("choose 'Smog Check Technician' from the 'License Type' dropdown")
  # v3 used page.locator("#licenseType >> option:checked"); v4 locator has
  # no ">>" chaining, so the same check is re-expressed in-page.
  selected_option = t.page.evaluate(<<~JS)
    (() => {
      const option = document.querySelector("#licenseType option:checked");
      return option?.textContent ?? null;
    })()
  JS

  if selected_option == "Smog Check Technician"
    next { _success: true }
  end

  { _success: false, message: "incorrect option selected from the dropdown" }
end
